pkg/wait/conditions.go

```diff
@@ -196,6 +207,85 @@ func IsAppVersion(ctx context.Context, kubeClient *client.Client, appName string
 	}
 }
 
+// IsClusterConditionSet returns a WaitCondition that checks if a Cluster resource has the specified condition with the expected status
+func IsClusterConditionSet(ctx context.Context, kubeClient *client.Client, clusterName string, clusterNamespace string, conditionType capi.ConditionType, expectedStatus corev1.ConditionStatus, expectedReason string) WaitCondition {
+	cluster := &capi.Cluster{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: clusterNamespace,
+		},
+	}
+
+	return isClusterApiObjectConditionSet(ctx, kubeClient, cluster, conditionType, expectedStatus, expectedReason)
+}
+
+// IsKubeadmControlPlaneConditionSet returns a WaitCondition that checks if a KubeadmControlPlane resource has the specified condition with the expected status
+func IsKubeadmControlPlaneConditionSet(ctx context.Context, kubeClient *client.Client, clusterName string, clusterNamespace string, conditionType capi.ConditionType, expectedStatus corev1.ConditionStatus, expectedReason string) WaitCondition {
+	kcp := &kubeadm.KubeadmControlPlane{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: clusterNamespace,
+		},
+	}
+
+	return isClusterApiObjectConditionSet(ctx, kubeClient, kcp, conditionType, expectedStatus, expectedReason)
+}
+
+// IsKubeadmControlPlaneConditionSet returns a WaitCondition that checks if a cluster has the specified condition with the expected status
+// Passing Kind explicitly here for the sake of more useful logs (as obj.GetObjectKind().GroupVersionKind().Kind returns empty string sometimes)
```

Comment:
Is this still correct? I don't see kind being passed in

Reply:
Right, I changed it in the meantime. Will fix.